Krypt/Source/SHA256.swift

```diff
@@ -61,4 +69,34 @@ public struct SHA256 {
 
     return digest
   }
+
+  /// iOS 13 Variant of SHA256 hash calculation for large files using a buffer and CryptoKit
+  /// - Parameters:
+  ///   - url: The url to the file to calculate the SHA256 hash for
+  ///   - withBufferSize: The size of the buffer to use in bytes, defaults to 1024*1024 bytes = 1 MB
+  @available(iOS 13, *)
+  public static func digestV2(file url: URL, withBufferSize: Int = 1024 * 1024) throws -> Data {
+    guard let stream = InputStream(fileAtPath: url.path) else {
+      throw SHA256Error.fileOperationError
+    }
+    stream.open()
+    let buffer = UnsafeMutablePointer<UInt8>.allocate(capacity: withBufferSize)
+    defer {
+      buffer.deallocate()
+    }
+    var hasher = CryptoKit.SHA256()
+    while stream.hasBytesAvailable {
+      let read = stream.read(buffer, maxLength: withBufferSize)
```

Comment:
Also this one still haunts me, you changed it from `with bufferSize: Int` to `withBufferSize: Int` while I was suggesting `withBufferSize bufferSize: Int` which gives you the best of both.
```suggestion
      let read = stream.read(buffer, maxLength: bufferSize)
```

Reply:
my bad will fix it